lib/modules/versioning/debian/index.spec.ts

```diff
@@ -0,0 +1,368 @@
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'foobar'}   | ${null}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'bullseye'} | ${'bullseye'}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'buster'}   | ${'buster'}
+    ${['jessie', 'stretch', 'buster', 'stable']}        | ${'stable'}   | ${'stable'}
+    ${['jessie', 'stretch', 'oldstable', 'bullseye']}   | ${'buster'}   | ${'oldstable'}
+    ${['jessie', 'oldoldstable', 'buster', 'bullseye']} | ${'warty'}    | ${null}
+  `(
+    'getSatisfyingVersion($versions, "$range") === "$expected"',
+    ({ versions, range, expected }) => {
+      expect(debian.getSatisfyingVersion(versions, range)).toBe(expected);
+    }
+  );
+
+  it.each`
+    versions                                            | range         | expected
+    ${['8', '9', '10', '11']}                           | ${'2020.04'}  | ${null}
+    ${['8', '9', '10', '11']}                           | ${'foobar'}   | ${null}
+    ${['8', '9', '10', '11']}                           | ${'11'}       | ${'11'}
+    ${['8', '9', '10', '11']}                           | ${'10'}       | ${'10'}
+    ${['8', '9', '10', '11']}                           | ${'4'}        | ${null}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'2020.04'}  | ${null}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'foobar'}   | ${null}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'bullseye'} | ${'bullseye'}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'buster'}   | ${'buster'}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'warty'}    | ${null}
+    ${['jessie', 'stretch', 'buster', 'stable']}        | ${'stable'}   | ${'stable'}
+    ${['jessie', 'stretch', 'oldstable', 'bullseye']}   | ${'buster'}   | ${'oldstable'}
+    ${['jessie', 'oldoldstable', 'buster', 'bullseye']} | ${'warty'}    | ${null}
+  `(
+    'minSatisfyingVersion($versions, "$range") === "$expected"',
+    ({ versions, range, expected }) => {
+      expect(debian.minSatisfyingVersion(versions, range)).toBe(expected);
+    }
+  );
+
+  it.each`
+    currentValue      | rangeStrategy | currentVersion | newVersion    | expected
+    ${undefined}      | ${undefined}  | ${undefined}   | ${'foobar'}   | ${'foobar'}
+    ${'stretch'}      | ${undefined}  | ${undefined}   | ${'11'}       | ${'bullseye'}
+    ${'stretch'}      | ${undefined}  | ${undefined}   | ${'bullseye'} | ${'bullseye'}
+    ${'stretch'}      | ${undefined}  | ${undefined}   | ${'stable'}   | ${'bullseye'}
+    ${'9'}            | ${undefined}  | ${undefined}   | ${'11'}       | ${'11'}
+    ${'oldoldstable'} | ${undefined}  | ${undefined}   | ${'11'}       | ${'stable'}
+    ${'oldstable'}    | ${undefined}  | ${undefined}   | ${'11'}       | ${'stable'}
+    ${'9'}            | ${undefined}  | ${undefined}   | ${'stable'}   | ${'11'}
+    ${'oldstable'}    | ${undefined}  | ${undefined}   | ${'3'}        | ${'3'}
```

Comment:
Why it should update to `3`?
```suggestion
    ${'oldstable'}    | ${undefined}  | ${undefined}   | ${'3'}        | ${'3'}
    ${'oldstable'}    | ${'pin'}            | ${undefined}   | ${'10'}        | ${'10'}
    ${'bullseye'}     | ${'pin'}            | ${undefined}   | ${'11'}        | ${'11'}
```

I think we should support pin range strategy

Reply:
> Why it should update to `3`?

it shouldn't, the test case is made only to cover some branches for coverage.
see -
https://github.com/renovatebot/renovate/pull/14881#discussion_r851712799

Reply:
OK, what about the two new test cases? 🙃